bsp: am6x: Add phyCORE-AM62L FPSC platform

FPlohl opened the pull request:
This PR introduces documentation support for the phyCORE-AM62L FPSC platform.

### Changes

- Copied files that can be shared between TI and NXP platforms from `imx-common` to bsp/ directory to avoid duplication.
- Added `ti-common` directory for TI specific documentation.
- Added `linux-phytec-ti` link stub to `conf.py`.
- Created **HEAD BSP manual for AM62L** with relevant documentation included.

### Notes on Common Documentation

Some content of "common" documentation is **i.MX specific**. There are multiple solutions to this problem but we did not find an elegant way to solve this. Ideally we could conditionally include content based on platform (similar to `#ifdef` in C). The closest thing we found for RST is the `only` directive, but it works only at build time. That means we would have to build seperatly for NXP and TI, and then merge the documentation, which isn't practical.

One such problematic part is the NXP EULA which we currently excluded using `end-before`/`start-after` statements. 

We'd appreciate feedback on how to handle these cross-platorm documentation challenges more elegantly.

TO-DO:

- [ ] Set ${overlays} variable
- [ ] U-Boot External Environment
- [ ] Port all imx documentation to use new common include paths

ymoog commented on source/bsp/development/host_network_setup.rsti:
```diff
@@ -150,15 +150,3 @@ When you boot/restart your host PC and don't have the network interface, as
-      u-boot=> env print ip_dyn
-      ip_dyn=no
-
-   To use dynamic IP addresses for netboot, ``ip_dyn`` needs to be set to ``yes``.
```
You are removing common code without replacement. Please give a reason why this is okay.

andrejpicej replied:
The idea behind this was that we now use dynamic IP for network boot with ampliphy-boot. I just recently noticed Benjamins u-boot patches after which we will probably support both. So I think this can be kept in. Also not sure if we now aggread that static IP or DHCP will be used in all releases?

ymoog commented on source/bsp/building-bsp.rsti:
```diff
@@ -86,6 +86,7 @@ Starting the Build Process
       This needs to be done every time you open a new shell for starting builds.
 
 *  The current working directory of the shell should change to build/.
+.. nxp-eula-marker
```
I think we should drop this and this should be a problem of i.MX manuals how they want to deal with this (i.e. informing readers of the FSL_EULA). The goal is to have the common code as generic as possible and this looks like an erroneous "forgotten" artifact.

andrejpicej replied:
Is this not needed anymore for newer NXP releases? Anyway, I see that this is also mentioned in yocto manuals.

ymoog replied:
It is needed. What I am trying to convey is that this should be in source/bsp/<imx> not in /source/bsp for the sake of maintaining separation of common and imx specific parts.

ymoog commented on source/bsp/ti-common/development/netboot.rsti:
What is the difference between the ti-common files and imx?
Is it possible to merge them into one generic rsti (for each ti-common files you introduce in this commit)?

To me it is important to know where the differences are. So that as a next step we can work on getting rid of the differences.

dominiknh90 replied:
With the upcoming AM68 and AM69 in mind it makes sense for me to split it up since they can also load the bootloader over network.

FPlohl replied:
> What is the difference between the ti-common files and imx? Is it possible to merge them into one generic rsti (for each ti-common files you introduce in this commit)?
> 
> To me it is important to know where the differences are. So that as a next step we can work on getting rid of the differences.

Bulk of the content is pretty much the same so they can be merged into one generic rsti. A lot of the differences are not single line so we cannot use `replace` for platform specific documentation. 

We could use something like this:
- `generic.rsti` – contains the shared content with markers where platform differences occur.
- `imx.rsti` and `ti.rsti` – contain platform specific documentation and include `generic.rsti` with multiple `start-after/end-before` statements.
- In the board documentation we then include only the `imx.rsti` or the `ti.rsti`.

ymoog replied:
I was hoping you could generate a diff easily. Anyway, I don't expect you to change this now.

dominiknh90 commented on source/bsp/am6x/am62l/bootmode-switch.rsti:
eMMC here is actually DEVBOOT which only got eFused to eMMC UDA. we need to update this for PD25 with the new hardware

dominiknh90 commented on source/bsp/am6x/peripherals/eeprom.rsti:
```diff
@@ -0,0 +1,50 @@
+.. code-block:: console
+
+   target:~$ dd if=/sys/class/i2c-dev/i2c-0/device/0-0050/eeprom bs=1 count=1024  | od -x
+
```
this is configured as I2C1 by devicetree alias for FPSC

dominiknh90 commented on source/bsp/am6x/peripherals/tm.rsti:
we'll also have to make a split for imx common and ti common here

dominiknh90 commented on source/bsp/am6x/am62l/head.rst:
```diff
@@ -0,0 +1,524 @@
+
+.. General Substitutions
+.. |doc-id| replace:: Head
+.. |kit| replace:: **phyCORE-AM62L FPSC Kit**
```
this will be phyFLEX-AM62L in the future

dominiknh90 commented on source/bsp/ti-common/development/standalone_build_u-boot_binman.rsti:
```diff
@@ -0,0 +1,120 @@
+.. hint::
+   The specific offset values are also declared in the Yocto variables
+   "BOOTLOADER_SEEK" and "BOOTLOADER_SEEK_EMMC"
+
```
we have the ti-boot-container binary which is the 3 binaries glued together for this use case

leitzgen reviewed (approved):
Looks good.

dominiknh90 commented on source/bsp/peripherals/eeprom.rsti:
```diff
@@ -1,17 +1,27 @@
+
+.. warning::
+
+   The first 256 (0x100) bytes of the EEPROM area (bus: I2C-|eeprom-detect-bus|
```
256 bytes is not enough for api v3 with block extensions. we'll need a soc variable here (for am62l it's 1024 byte)

dominiknh90 commented on source/bsp/peripherals/eeprom.rsti:
```diff
@@ -56,8 +66,38 @@ or:
+   The first 256 (0x100) bytes of the EEPROM area (bus: I2C-|eeprom-detect-bus|
+   addr: |eeprom-detect-addr|) should not be erased or overwritten. As this will
+   influence the behavior of the bootloader. The board might not boot correctly
+   anymore.
```
same as above

dominiknh90 reviewed (changes requested):
Hi Florijan,
I found a few more things to change.

In addition we'll have to rework the common eMMC page. IMX uses UDA with offset to boot and we'll have both UDA with FAT and HW boot partition (with offset 0) support in the PD25

FPlohl commented on source/bsp/am6x/am62l/head.rst:
```diff
@@ -0,0 +1,517 @@
+'SS'). The USB subsystem has two independent USB controller cores. Both cores
+are capable of acting as a USB peripheral device or a USB host. Each is
+connected to a USB 3.0 PHY.
+
```
Corrected this paragraph with accurate information

dominiknh90 reviewed (approved):
There are more open todos/fixes but I'm happy to address those in a new pull request as this one is already quite large.

BHahn42618 commented on source/bsp/development/master_manifest.rsti:
```diff
@@ -0,0 +1,16 @@
+-------------------------------------
+
+We have a vanilla manifest that makes use of the Yocto master branches (not an
+TI release), Linux, and U-Boot. This can be used to test the latest upstream
```
How about "vendor release)"? To make this general and not TI/NXP specific.

BHahn42618 commented on source/bsp/peripherals/network.rsti:
```diff
@@ -0,0 +1,120 @@
+``meta-ampliphy/recipes-core/systemd/systemd-conf``.
+
+IP addresses can be configured within \*.network files. The interfaces are
+configured to static IP as default. The default IP address and netmask for end0
```
Is this end style something common for all BSPs, or something TI specific? Otherwise you could move this to ti-common folder.

BHahn42618 commented on source/bsp/ti-common/development/netboot.rsti:
```diff
@@ -38,47 +33,42 @@ Place Images on Host for Netboot
+.. code-block:: console
 
-   All supported devicetree overlays are in the |ref-dt| chapter.
+   u-boot=> setenv boot_targets dhcp
```
Are you sure this is correct? I believe the boot_target for standardboot is ethernet not dhcp.

FPlohl commented on source/bsp/peripherals/eeprom.rsti:
```diff
@@ -1,17 +1,27 @@
+
+.. warning::
+
+   The first |eeprom-som-detect-area| (|eeprom-som-detect-area-hex|) bytes of
```
They should work. imx manuals don't include this file yet, they still use `source/bsp/imx9/peripherals/eeprom.rsti`, `source/bsp/imx8/peripherals/eeprom.rsti` or `/bsp/imx-common/peripherals/eeprom.rsti`

FPlohl commented:
I started working on AM68 documentation and I realised some changes needed to be made to common documentation. Notable changes: 
- moved `source/bsp/ti-common/gpio.rsti` to `source/bsp/peripherals/gpio.rsti` and moved board specific GPIO information to `am62l/head.rst`
- moved **Get the needed binaries** chapter from `source/bsp/ti-common/development/standalone_build_u-boot_binman.rsti` to `am62l/head.rst`
- moved `mtdinfo --all` code snippet from `source/bsp/am6x/peripherals/spi-nor-flash-no-boot.rsti` to `am62l/head.rst`
